### PR TITLE
fix: [PL-21765]: Added proper Error messages when Request is not containing the mandatory fields.

### PR DESCRIPTION
**Pull request by boopesh007**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31260)

### CODE REVIEW

**Comment by boopesh007** [comment]
trigger messagecheck

**Comment by boopesh007** [comment]
trigger codeformat

**Comment by boopesh007** [comment]
trigger checkstyle

File: 970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java [hunk 1]
```diff
@@ -21,11 +20,17 @@ public String getFullyQualifiedIdentifier(
       throw new InvalidRequestException("Invalid Request Exception: No identifier provided.");
     }
     if (EmptyPredicate.isNotEmpty(projectIdentifier)) {
-      Asserts.notEmpty(orgIdentifier, "Org Identifier");
-      Asserts.notEmpty(accountId, "Account Identifier");
+      if (EmptyPredicate.isEmpty(orgIdentifier)) {
+        throw new InvalidRequestException("Invalid Request Exception: No org identifier provided.");
```
Comment (meenakshi-harness) [inline 970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java:24]: I think `Invalid Request Exception` is not required.

File: 970-ng-commons/src/test/java/io/harness/utils/FullyQualifiedIdentifierHelperTest.java [hunk 2]
```diff
@@ -44,12 +43,23 @@ public void getFullyQualifiedIdentifier() {
...
+  @Test
+  @Owner(developers = OwnerRule.BOOPESH)
+  @Category(UnitTests.class)
+  public void testNoOrgIdentiferForPrjectConnectorThrowsException() {
+    String connectorIdentifier = randomAlphabetic(7);
+    assertThatThrownBy(()
+                           -> FullyQualifiedIdentifierHelper.getFullyQualifiedIdentifier(
+                               randomAlphabetic(10), null, randomAlphabetic(8), connectorIdentifier))
+        .isInstanceOf(InvalidRequestException.class);
+  }
```
Comment (meenakshi-harness) [inline 970-ng-commons/src/test/java/io/harness/utils/FullyQualifiedIdentifierHelperTest.java:64]: Can we add case for :
1. ProjectIdentifier is not null, but orgId is null. 
2. ProjectIdentifier is not null, but accountId is null. 
3. OrgId level entity --> but accountId is null

File: 970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java (hunk 1 above)
Comment (boopesh007) [reply 970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java:24]: yes, @meenakshi-harness felt that but added as the reference had it. will change it.

File: 970-ng-commons/src/test/java/io/harness/utils/FullyQualifiedIdentifierHelperTest.java (hunk 2 above)
Comment (boopesh007) [reply 970-ng-commons/src/test/java/io/harness/utils/FullyQualifiedIdentifierHelperTest.java:64]: 1. Is that my test checks 
2. I will add 
3. already added by Naman

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java, line 31 at r3](https://reviewable.io/reviews/harness/harness-core/31260#-MxfZpUl96LlvzQx23KJ:-MxfZpUl96LlvzQx23KK:b3uojgz) ([raw file](https://github.com/harness/harness-core/blob/9008e19dc1c675d44e8fdd0f4b2fed1618854815/970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java#L31)):*
> ```Java
>       return String.format("%s/%s/%s/%s", accountId, orgIdentifier, projectIdentifier, identifier);
>     } else if (EmptyPredicate.isNotEmpty(orgIdentifier)) {
>       if (EmptyPredicate.isEmpty(accountId)) {
> ```

I see repetition of code. We can modularize check for account id and use it at both places as function call only. We can do this for org id check also, even though its called only once.

Method could be a void function like: validateAccountId() and it will throw exception if validation fails.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java, line 33 at r4](https://reviewable.io/reviews/harness/harness-core/31260#-MxhRN3F1mXHwq76nuBO:-MxhRN3G-_Au0y5dvRW2:b-ysa4lo) ([raw file](https://github.com/harness/harness-core/blob/3edff3eb23ee1d27bde89b3b5c11aad899e9c634/970-ng-commons/src/main/java/io/harness/utils/FullyQualifiedIdentifierHelper.java#L33)):*
> ```Java
>       throw new InvalidRequestException("No identifier provided.");
>     }
>   }
> ```

missing newline after this method

**Review by mohitg0795 — DISMISSED** [review]
:lgtm:

> Reviewed 1 of 2 files at r3, 1 of 2 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260#-:-MxhTL2B3ktSbVHDQDqe:bnfp4nl)* status: 3 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by meenakshi-harness — COMMENTED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260)* status: 1 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by mohitg0795 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31260#-:-MxhmUJ-EeCCAJJwJDbT:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**